### PR TITLE
(SIMP-3753) simp-utils Add YAML-based ENC script

### PR DESCRIPTION
**Pull request by lnemsick-simp**
SIMP-3753 #close
SIMP-3754 #close
SIMP-3755 #close
SIMP-3486 #close

### CODE REVIEW

File: scripts/bin/set_environment [hunk 1]
```diff
@@ -0,0 +1,130 @@
+#!/bin/ruby
```
Comment (jeefberkey) [inline scripts/bin/set_environment]: Can we try to run this using a shebang pointing to `/opt/puppetlabs/puppet/bin/ruby`?

Comment (lnemsick-simp) [reply scripts/bin/set_environment]: I was unable to get that to work on our dev environment, when I was working on setting up R10K.  The script worked when called outside of the Puppet server, but failed when called by the Puppet server.  I'll revisit to see if I can figure out what the issue was.

Comment (lnemsick-simp) [reply scripts/bin/set_environment]: The problem on our dev Puppet master is that many of the gems in /opt/puppetlabs/puppet are not accessible to puppet.  So, in the puppet server log, you get something like:
 WARN  [c.p.p.ShellUtils] Executed an external process which logged to STDERR: /opt/puppetlabs/puppet/lib/ruby/2.1.0/rubygems/stub_specification.rb:71:in `initialize': Permission denied @ rb_sysopen - /opt/puppetlabs/puppet/lib/ruby/gems/2.1.0/specifications/blankslate-2.1.2.4.gemspec (Errno::EACCES)

Comment (jeefberkey) [reply scripts/bin/set_environment]: For now, I think it's fine to leave it as `#!/bin/ruby`. There may be a future ticket where we are interested in not installing system ruby by default.

Comment (trevor-vaughan) [reply scripts/bin/set_environment]: The permissions problems are probably that someone forgot to change their `umask` before installing the gems.

**Pull request merged**